### PR TITLE
add basic CriiptoSignaturesClient constructor

### PR DESCRIPTION
**Pull request by mickhansen**
@sgryt As I'm not really an expert on idiomatic C#/dotnet I'll send a few PRs your way just as a litmus test :)

### CODE REVIEW

File: Criipto.Signatures/Client.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    public class CriiptoSignaturesClient : IDisposable {
+        private GraphQLHttpClient graphQLClient;
+
+        public CriiptoSignaturesClient(string clientId, string clientSecret, string endpoint = "https://signatures-api.criipto.com/v1/graphql") {
+            graphQLClient = new GraphQLHttpClient(endpoint, new NewtonsoftJsonSerializer());
+            graphQLClient.HttpClient.DefaultRequestHeaders.Authorization
+                = new AuthenticationHeaderValue("Basic", Convert.ToBase64String(System.Text.Encoding.ASCII.GetBytes($"{clientId}:{clientSecret}")));
+        }
+
+        public void Dispose() {
```
Comment (sgryt) [inline Criipto.Signatures/Client.cs]: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1063

File: Criipto.Signatures/Client.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+
+using GraphQL.Client.Abstractions;
+using GraphQL.Client.Http;
+using GraphQL.Client.Serializer.Newtonsoft;
+using System.Net.Http.Headers;
+
+namespace Criipto.Signatures {
+    public class CriiptoSignaturesClient : IDisposable {
+        private GraphQLHttpClient graphQLClient;
```
Comment (sgryt) [inline Criipto.Signatures/Client.cs]: Making members `private readonly` is idiomatic

**Review by sgryt — CHANGES_REQUESTED** [review]
And because C# is so wildly inconsistent, enabling "warnings as errors" in the compilation settings for all projects is good practice. As well as running code analysis frequently, and not defaulting to just ignoring the warnings.

File: Criipto.Signatures/Client.cs (hunk 2 above)
Comment (mickhansen) [reply Criipto.Signatures/Client.cs]: @sgryt `readonly` is settable from constructor?

**Comment by mickhansen** [comment]
> And because C# is so wildly inconsistent, enabling "warnings as errors" in the compilation settings for all projects is good practice. As well as running code analysis frequently, and not defaulting to just ignoring the warnings.

"warnings as errors" might be difficult in this project, the graphql to C# codegen generates typings with quite a few warnings.

File: Criipto.Signatures/Client.cs (hunk 2 above)
Comment (sgryt) [reply Criipto.Signatures/Client.cs]: Yes, but not mutable from other members.

**Comment by sgryt** [comment]
> > And because C# is so wildly inconsistent, enabling "warnings as errors" in the compilation settings for all projects is good practice. As well as running code analysis frequently, and not defaulting to just ignoring the warnings.
> 
> "warnings as errors" might be difficult in this project, the graphql to C# codegen generates typings with quite a few warnings.

It is possible to selectively mark generated code as "ignore warnings"

File: Criipto.Signatures/Client.cs (hunk 1 above)
Comment (mickhansen) [reply Criipto.Signatures/Client.cs]: @sgryt Hmm I'm not getting a warning for this rule.

Comment (mickhansen) [reply Criipto.Signatures/Client.cs]: Nvm, had to add `<AnalysisMode>All</AnalysisMode>`

**Comment by mickhansen** [comment]
@sgryt Updated to include warnings as errors and code analysis

File: Criipto.Signatures.UnitTests/CriiptoSignaturesClientTests.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+using Xunit;
+
+namespace Criipto.Signatures.UnitTests;
+
+public class CriiptoSignaturesClientTests
+{
+    [Fact]
+    public void IsDisposable()
```
Comment (sgryt) [inline Criipto.Signatures.UnitTests/CriiptoSignaturesClientTests.cs:8]: Suggest to change to `using` syntax, that will make the compiler check that `CriiptoSignaturesClient` implements `IDisposable` - the current construct does not guarantee that.

**Comment by sgryt** [comment]
I'm probably missing some local setup here, but a WSL-based .sln build from VS Code gives me compilation errors:
image

**Comment by mickhansen** [comment]
> I'm probably missing some local setup here, but a WSL-based .sln build from VS Code gives me compilation errors: image

Aye you need to install the SDK for .NET 7

File: Criipto.Signatures.UnitTests/CriiptoSignaturesClientTests.cs (hunk 3 above)
Comment (mickhansen) [reply Criipto.Signatures.UnitTests/CriiptoSignaturesClientTests.cs:8]: @sgryt Still manually dispose, or?

Comment (sgryt) [reply Criipto.Signatures.UnitTests/CriiptoSignaturesClientTests.cs:8]: No, not needed. The `using` statement takes care of that.

**Comment by sgryt** [comment]
> > I'm probably missing some local setup here, but a WSL-based .sln build from VS Code gives me compilation errors: image
> 
> Aye you need to install the SDK for .NET 7

But the readme says .NET 6 is supported ?
```
### Requirements

This library supports .NET 6 and .NET 7.
```

**Comment by sgryt** [comment]
Hmm, my WSL does not want to install .NET 7. Trying on git bash now.

**Comment by mickhansen** [comment]
> > > I'm probably missing some local setup here, but a WSL-based .sln build from VS Code gives me compilation errors: image
> > 
> > 
> > Aye you need to install the SDK for .NET 7
> 
> But the readme says .NET 6 is supported ?
> 
> ```
> ### Requirements
> 
> This library supports .NET 6 and .NET 7.
> ```

Aye, I believe you will be able to install it via nuget for .NET 6, but compiling it requires .NET 7 SDKs.

**Comment by mickhansen** [comment]
> Hmm, my WSL does not want to install .NET 7. Trying on git bash now.

I was able to do it via `apt-get install`

**Comment by sgryt** [comment]
@mickhansen Last 3 commits try to illustrate my understanding of idiomatic C# - but I admit it has been a while since I have been concerned with that...

**Comment by mickhansen** [comment]
> @mickhansen Last 3 commits try to illustrate my understanding of idiomatic C# - but I admit it has been a while since I have been concerned with that...

@sgryt Seems reasonable to me :)

**Pull request merged**